src/main/java/core/basesyntax/service/impl/PurchaseAction.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ActionHandler;
+
+public class PurchaseAction implements ActionHandler {
+    public void apply(String fruit, int quantity) {
+        Storage.removeFruit(fruit, quantity);
+    }
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. You need to implement a check to ensure the balance does not go negative after the purchase, and throw a RuntimeException if it does.